snuba/subscriptions/entity_subscription.py

```diff
@@ -0,0 +1,133 @@
+from abc import ABC
+from enum import Enum
+from typing import Any, List, Mapping, Optional, Sequence, Type, Union
+
+from snuba.datasets.entities import EntityKey
+from snuba.query.conditions import ConditionFunctions, binary_condition
+from snuba.query.exceptions import InvalidQueryException
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.types import Condition
+
+
+class SubscriptionType(Enum):
+    LEGACY = "legacy"
+    SNQL = "snql"
+    DELEGATE = "delegate"
+
+
+class EntitySubscription(ABC):
+    def __init__(
+        self, subscription_type: SubscriptionType, data_dict: Mapping[str, Any]
+    ) -> None:
+        self.subscription_type = subscription_type
+
+    def get_entity_subscription_conditions(
+        self, offset: Optional[int] = None
+    ) -> Sequence[Union[Expression, Condition]]:
+        return []
+
+    def to_dict(self) -> Mapping[str, Any]:
+        return {}
```

Comment:
Why though? I don't want this function to be mandatory and I want this to be the default behaviour if I don't define the function

Reply:
I guess it just seems unlikely to me that in practice we will have any entities which actually have that default behaviour, which defeats the point of making it default. To me, the downside of the default is that it makes it less clear to the next person providing a new EntitySubscription subclass whether they need to provide an implementation or use the default, and it's easier to just forget to provide it since it won't throw an error.